crates/polars-plan/src/plans/builder_ir.rs

```diff
@@ -302,4 +302,65 @@ impl<'a> IRBuilder<'a> {
         };
         self.add_alp(lp)
     }
+
+    pub fn sink(self, payload: SinkType) -> PolarsResult<Self> {
+        // This is duplicated from `dsl_to_ir::to_alp_impl`, would be good to refactor this and deduplicate.
```

Comment:
Can we do that in this PR? 

Make a function and factor out the shared arguments?